tests/test_bgp_af.rb

```diff
@@ -390,123 +440,78 @@ def test_dampen_igp_metric
+    assert_equal(bgp_af.default_dampen_igp_metric, bgp_af.dampen_igp_metric,
                  "Error: Default 'dampen-igp-metric' value should be " \
                    "#{bgp_af.default_dampen_igp_metric}")
 
-    #
-    # Set and verify 'dampen-igp-metric <value>'
-    #
-
-    # Do a 'dampen-igp-metric 555'
-    pattern = /^ *dampen-igp-metric 555$/
-    bgp_af.dampen_igp_metric = 555
-
-    af_string = get_bgp_af_cfg(asn, vrf, af)
-
-    assert_match(pattern, af_string,
-                 "Error: 'dampen-igp-metric 555' is not configured " \
-                   'and should be')
-    #
-    # Test getter
-    #
-    assert_equal(bgp_af.dampen_igp_metric, 555,
-                 'Error: dampen_igp_metric should be 555')
-
-    #
-    # Set and verify 'no dampen-igp-metric'
-    #
+    if platform == :ios_xr
+      assert_raises(Cisco::UnsupportedError) do
+        bgp_af.dampen_igp_metric = 555
+      end
+    else
+      # Set and verify 'dampen-igp-metric <value>'
+      bgp_af.dampen_igp_metric = 555
 
-    # Do a 'no dampen-igp-metric'
-    pattern = /no dampen-igp-metric$/
-    bgp_af.dampen_igp_metric = nil
+      # Test getter
+      assert_equal(555, bgp_af.dampen_igp_metric,
+                   'Error: dampen_igp_metric should be 555')
 
-    af_string = get_bgp_af_cfg(asn, vrf, af)
+      # Set and verify 'no dampen-igp-metric'
+      bgp_af.dampen_igp_metric = nil
 
-    assert_match(pattern, af_string,
-                 "Error: 'no dampen-igp-metric' is not configured " \
-                   'and should be')
-    #
-    # Test getter
-    #
-    assert_equal(bgp_af.dampen_igp_metric, nil,
-                 'Error: dampen_igp_metric should be nil')
+      # Test getter
+      assert_equal(nil, bgp_af.dampen_igp_metric,
+                   'Error: dampen_igp_metric should be nil')
 
-    #
-    # Set default value explicitly
-    #
-    bgp_af.dampen_igp_metric = bgp_af.default_dampen_igp_metric
-    assert_equal(bgp_af.dampen_igp_metric, bgp_af.default_dampen_igp_metric,
-                 "Error: Default 'dampen-igp-metric' value should be " \
-                   "#{bgp_af.default_dampen_igp_metric}")
+      # Set default value explicitly
+      bgp_af.dampen_igp_metric = bgp_af.default_dampen_igp_metric
+      assert_equal(bgp_af.default_dampen_igp_metric, bgp_af.dampen_igp_metric,
+                   "Error: Default 'dampen-igp-metric' value should be " \
+                     "#{bgp_af.default_dampen_igp_metric}")
+    end
   end
 
   ##
-  ## dampening
+  ## Negative tests with non-default vrf on ios_xr
   ##
-  # rubocop:disable Metrics/MethodLength
-  def test_dampening
+  def dampening_vrf_xr_neg_test
     asn = '101'
+    vrf = 'red'
+    af = %w(ipv4 unicast)
+    bgp_af = RouterBgpAF.new(asn, vrf, af)
+    assert_nil(bgp_af.dampening,
+               'dampening should return nil on XR with non-default vrf')
+    assert_raises(Cisco::UnsupportedError) do
+      bgp_af.dampening = %w(one two)
+    end
```

Comment:
bgp_af.destroy ? Can you check you're doing this consistently for all test methods. Also in some you add an extra line before it and some you don't - can you make it consistent please?